test/unit/services/test_image.py

```diff
@@ -160,3 +161,48 @@ def test_create_with_invalid_entity_id(self):
         self.mock_create({**IMAGE_POST_METADATA_DATA_ALL_VALUES, "entity_id": "invalid-id"})
         self.call_create_expecting_error(InvalidObjectIdError)
         self.check_create_failed_with_exception("Invalid ObjectId value 'invalid-id'")
+
+
+class ListDSL(ImageServiceDSL):
+    """Base class for `list` tests."""
+
+    _entity_id_filter: Optional[str]
+    _primary_filter: Optional[str]
+    _expected_images: List[ImageSchema]
+    _obtained_images: List[ImageSchema]
+
+    def mock_list(self, image_in_data: list[dict]) -> None:
+        """Mocks repo methods appropriately to test the `list` service method."""
+
+        self._expected_images = [
+            ImageSchema(**ImageOut(**ImageIn(**image_in_data).model_dump()).model_dump())
+            for image_in_data in image_in_data
+        ]
+        self.mock_image_repository.list.return_value = [
+            ImageOut(**ImageIn(**image_in_data).model_dump()) for image_in_data in image_in_data
+        ]
```

Comment:
```suggestion
    def mock_list(self) -> None:
        """Mocks repo methods appropriately to test the `list` service method."""

        # Just returns the result after converting it to the schemas currently, so actual value doesn't matter here
        images_out = [ImageOut(**ImageIn(**IMAGE_IN_DATA_ALL_VALUES).model_dump())]
        self.mock_image_repository.list.return_value = images_out
        self._expected_images = [ImageSchema(**image_out.model_dump()) for image_out in images_out]
```
I would just do something like this, the actual value doesnt matter the same as the MagicMocks on IMS. As long as the data is valid it will test the result is returned after going through the schema. Then the test function doesn't even need to provide this.